nemo/collections/asr/modules/transformer/transformer_decoders.py

```diff
@@ -84,6 +103,15 @@ def forward_preln(self, decoder_query, decoder_mask, decoder_keys, encoder_state
         output_states = self.third_sub_layer(enc_dec_attn_output)
         output_states += residual
 
+        if self.is_adapter_available():
```

Comment:
looks like this is outside of residual connection. making sure that this is what is intended.

Reply:
Yes, linear adapter (determined by contents of pack_input) should run after the residual (it has its own residual inside it, and is zero init to prevent model actvations from shifting during init).